app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -221,6 +222,7 @@ class WireActivityViewModel @Inject constructor(
         }
 
         viewModelScope.launch {
+            delay(SHORT_DELAY) // to make sure that the app is fully initialized before handling the import intent
```

Comment:
I'm afraid that delay doesn't ensure that the order of actions will always be correct, the issue here is probably that `handleDeepLink` happens before we subscribe to the navigation flow, and we can't use `replay` to get over it because we had replay already and it was causing other problems. I suggest that we should either update `startDestination` or move `onComplete` to be called when the flow is subscribed:
```
        LaunchedEffect(scope) {
            navigationManager.navigateState
                .onSubscription { onComplete() }
                .onEach { ...
```
Second option is easier and I can't think of any drawbacks of this solution, what do you think?

Reply:
That's it! I didn't really know why the navigation was behaving so funky, but you were right, the navigate to import screen command was being ignored as the graph wasn't subscribed at that point. I made the changes and pushed and now it works like a charm! Thx again 🙇

Reply:
@gongracr dop not forget to change the PR title ;P